include/lsst/afw/geom/detail/frameSetUtils.h

```diff
@@ -94,14 +94,30 @@ for two purposes:
 - Read the approximate FITS WCS that LSST writes to FITS images (for use by non-LSST code).
 
 The frames of the returned WCS will be as follows:
-- base: an ast::Frame with domain "PIXELS0": pixels with 0,0 the lower left corner of the parent image
-- current: an ast::SkyFrame: ICRS RA, Dec
+- "PIXELS" (base frame): pixel frame with 0,0 the lower left corner of the image (LSST convention)
+- "IWC": FITS-WCS intermediate world coordinate systems
+- "SKY" (current frame): an ast::SkyFrame with domain "SKY": ICRS RA, Dec
+
+All frames are instances of ast::Frame except the SKY frame. All have 2 axes.
 
 @param[in,out] metadata  FITS header cards
-@param[in] strip  If true strip items from `metadata` that are used to create the WCS,
-    excluding `LTV1` and `LTV2`, which are always retained.
+@param[in] strip  If true: strip items from `metadata` used to create the WCS;
+    always keep date-related keywords such as "DATE-OBS" and "TIMESYS" (but not "EQUINOX")
+    because they might be wanted for other purposes.
+
+@throws lsst::pex::exceptions::InvalidParameterError if the metadata does not describe a celestial WCS.
+*/
+std::shared_ptr<ast::FrameDict> readLsstSkyWcs(daf::base::PropertySet& metadata, bool strip = true);
+
+/**
+Copy values from an AST FitsChan into a PropertyList
+
+@warning COMMENT and HISTORY cards are treated as string values
+
+@throws lsst::pex::exceptions::InvalidParameterError if `fitsChan` contains cards whose
+    type is not supported by `PropertyList`: complex numbers, or cards with no value
```

Comment:
Most of these functions throw `InvalidParameterError` if the provided argument cannot be used for some reason. I suppose `TypeError` would work, too, though it may complicate error handling in a caller.